### DISCUSSION TITLE
Flows Event Hook Filter (Blocking) create empty user

### DISCUSSION THREAD

**Discussion by morey6757** [directus/directus#16537] [Archive]
Hello world !

i try to use Event Hook 'items.create' on Directus Users.

When i use 'Action (Non Blocking)' all it's OK. My user is created correctly.

But when i use 'Filter (Blocking)', my user is created ... empty !!!
![image](https://user-images.githubusercontent.com/6492250/202888684-61cd796b-c243-4943-9691-fe18373e82de.png)

i am on latest v9.21.0

The goal being to add a send email later ! (only if user is created correctly)

Thanks you for your help :-)

**Comment by br41nslug** [comment]
When using `filter (Blocking)` this intercepts the incoming data allowing you to modify it before saving. It sounds like you've not properly passed the payload on thus resulting in an empty user. Make sure the last item in your flow returns the proper "user create" payload.

**Comment by morey6757** [comment]
Thanks for your response @br41nslug :-)


If i understand, i must add an action after the 'Send Email'

![image](https://user-images.githubusercontent.com/6492250/204612538-a6abcb4e-ba80-4c0e-899c-01c35a0b979d.png)

**Comment by morey6757** [comment]
Thanks you @br41nslug !

It's OK :-)

i add an action (Run Script) for return the payload :-)

![image](https://user-images.githubusercontent.com/6492250/205446249-f27c8966-0d54-4361-974c-3e7742659fdc.png)
